src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,126 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int INCREASE_VALUE = 2;
+    private int size;
+    private float loadFactor;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
+
     @Override
     public void put(K key, V value) {
+        resize();
+        Node<K, V> node = new Node<>(hash(key), key, value, null);
+
+        int index = node.hash % table.length;
+        if (table[index] == null) {
+            table[index] = node;
+
+        } else {
+            Node<K, V> current = table[index];
+            Node<K, V> prev = null;
+            while (current != null) {
+                if (Objects.equals(current.key, node.key)) {
+                    current.value = node.value;
+                    return;
+                }
+                prev = current;
+                current = current.next;
+            }
+            prev.next = node;
+        }
+        size++;
 
     }
 
     @Override
     public V getValue(K key) {
+        int indexBuket = hash(key) % table.length;
```

Comment:
create method for getting index
```suggestion
        int indexBuket = getIndex(key);
```